perf(linter): use `binary_search` for arrays with more than `7` elements

shulaoda opened the pull request:
Related to #10076 

Based on recent conclusions, replace the use of `contains` with `binary_search` to improve lookup performance for collections where `phf_set` was replaced with `array` before.

Boshen commented:
### Merge activity

* **Apr 11, 1:39 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

shulaoda committed:
perf(linter): use `binary_search` for arrays with more than `7` elements (#10357)

Related to #10076

Based on recent findings, improve the lookup efficiency by replacing the `array` with a `phf_set`.
